tests/test_adapters.py

```diff
@@ -57,66 +75,145 @@ async def test_prompt_adapter():
 
 
 @pytest.mark.asyncio
-async def test_lora_adapter():
+async def test_lora_adapter(vllm_model_handler):
+    base_model_name = "/granite/granite-3b-base-v2/step_75000_ckpt"
     adapter_name = "granite-3b-code-instruct-lora"
+    lora_path = str(FIXTURES_DIR) + "/" + adapter_name
     request = BatchedGenerationRequest(
         adapter_id=adapter_name,
+        model_id=base_model_name,
     )
+    mock_lora_requests = PropertyMock(
+        side_effect=[
```

Comment:
I expected this to be `return_value` instead of `side_effect` 🤔🤔🤔

(But not a problem if this works)

Reply:
so the name on this is really confusing. return value (to my understanding) only lets you specify a single return value which is always the same. for this test to work I need to be able to set multiple different return values in sequence to fake the expected behavior of vllm actually loading and caching the adapter. side_effect seems to just be a way that lets you specify multiple return values that supersede return_value even if it's set, which to me doesn't seem like a side effect but more of a primary effect? The naming me threw me for a loop when I was trying to figure this out.

Reply:
ah, yeah that sounds about right for python mocks!